platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/style/types/VariableAnchorOffset.kt

```diff
@@ -0,0 +1,81 @@
+package org.maplibre.android.style.types
+
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * Represents a collection of anchor offsets.
+ */
+class VariableAnchorOffset(values: Array<AnchorOffset>) {
+    private val anchorOffsets: Array<AnchorOffset>
+
+    init {
+        values.copyOf().also { anchorOffsets = it }
+    }
+
+    /**
+     * Converts the collection to a JSON string representation.
+     *
+     * @return The JSON string representation of the collection.
+     */
+    override fun toString(): String {
+        val jsonArray = JSONArray()
+        for (entry in anchorOffsets) {
+            val entryArray = JSONArray()
+            entryArray.put(entry.offset[0])
+            entryArray.put(entry.offset[1])
+
+            val entryObject = JSONObject()
+            entryObject.put(entry.anchorType, entryArray)
+
+            jsonArray.put(entryObject)
+        }
+
+        return jsonArray.toString()
+    }
+
+    /**
+     * Converts the collection to an array.
+     *
+     * @return An array containing the anchor types and their offsets.
+     */
+    fun toArray(): Array<Any?> {
```

Comment:
Why is `Array<Any?>` used here?

Reply:
It is used in Layer.java
  private Object convertValue(@Nullable Object value) {
    if (value instanceof Expression) {
      return ((Expression) value).toArray();
    } else if (value instanceof Formatted) {
      return ((Formatted) value).toArray();
    **} else if (value instanceof VariableAnchorOffset) {
      return ((VariableAnchorOffset) value).toArray();
    }**
    else {
      return value;
    }
  }